packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -0,0 +1,323 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { assert, fromBase64ToUtf8, hashFile, IsoBuffer, Uint8ArrayToString } from "@fluidframework/common-utils";
+import { ISummaryContext } from "@fluidframework/driver-definitions";
+import { getGitType } from "@fluidframework/protocol-base";
+import * as api from "@fluidframework/protocol-definitions";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import {
+    IBlob,
+    ISnapshotRequest,
+    ISnapshotResponse,
+    ISnapshotTree,
+    ISnapshotTreeBaseEntry,
+    SnapshotTreeEntry,
+    SnapshotTreeValue,
+    SnapshotType,
+} from "./contracts";
+import { EpochTracker, FetchType } from "./epochTracker";
+import { getUrlAndHeadersWithAuth } from "./getUrlAndHeadersWithAuth";
+import { getWithRetryForTokenRefresh } from "./odspUtils";
+import { TokenFetchOptions } from "./tokenFetch";
+
+/* eslint-disable max-len */
+
+interface IDedupCaches {
+    blobShaToPath: Map<string, string>,
+    pathToBlobSha: Map<string, string>,
+    treesPathToTree: Map<string, api.ISummaryTree>,
```

Comment:
First two are rather clear in their intentions.
It would be great to add comment for treesPathToTree to indicate what kind of tree we are storing, i.e. is it always fully expanded tree? Does it content blob content or only blob IDs?

Reply:
Added comments.